pkg/controller/constraint/constraint_controller.go

```diff
@@ -160,12 +166,10 @@ func (r *ReconcileConstraint) Reconcile(request reconcile.Request) (reconcile.Re
 			return reconcile.Result{}, err
 		}
 		if _, err := r.opa.AddConstraint(context.Background(), instance); err != nil {
-			r.reportTotalConstraints(
-				constraintKey,
-				Tags{
-					enforcementAction: enforcementAction,
-					status:            errorStatus,
-				})
+			r.addConstraintKey(constraintKey, Tags{
```

Comment:
This doesn't trigger a report because it returns before the bottom of the function.

One pattern that may help:

create an `reportMetrics` boolean that gets set to true if metrics need to be updated and have this code:

```
defer func() {
   if reportMetrics {
      reportTotalConstraints()
   }
}()
```

This allows us to use defer to guarantee we try to report, but avoid unnecessary reporting.

Reply:
my bad, didn't realize. good idea about defer, thanks